File: Leet Code/01_Merge_alternate_string/01_merge_alternate_string.go
package main

import "fmt"

// word1 --> abcd
// word2 --> efghjk
// output --> aebfcgdhjk

func main() {
	fmt.Println(mergeAlternateString("abcd", "zxcvaa"))
}

func mergeAlternateString(word1, word2 string) string {
	var result string
	maxlength := len(word1)
	if len(word2) > maxlength {
		maxlength = len(word2)
	}
	for i := 0; i < maxlength; i++ {
		if i < len(word1) {
			result += string(word1[i])
		}

		if i < len(word2) {
			result += string(word2[i])
		}
	}
	return result
}

File: Leet Code/01_Merge_alternate_string/01_merge_alternate_string.py
def mergeAlternateString(word1,word2):
    result = ""
    l = len(word1)
    if len(word2) > l:
        l = len(word2)
    for x in range(0,l):
        if x <  len(word1):
            result += word1[x]
        if x <  len(word2):
            result += word2[x]  
    return result          


print(mergeAlternateString("abc","defgg"))